framework/include/utils/StaticallyAllocatedSet.h

```diff
@@ -0,0 +1,123 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "MooseError.h"
+
+#include <array>
+
+namespace MooseUtils
+{
+
+/**
+ * Optimized Set
+ *
+ * For when you need a small sized, really fast set that is small and doesn't do any allocation
+ *
+ * *Small* is important here!  The search routine is _linear_!
+ *
+ * It is templated on the type the set will hold and the maximum size of the set (N)
```

Comment:
This description should include the fact that the buffer only holds a unique copy of each element